### DISCUSSION TITLE
Automatically Wait For Anthropic Rate Limit (rate_limit_error)

### DISCUSSION THREAD

**Discussion by garrett-hopper** [cline/cline#871] [Feature Requests]
I would like Cline to automatically detect and wait to retry these errors. (Or even track token usage and wait before hitting them in the first place.)

```
429 {"type":"error","error":{"type":"rate_limit_error","message":"This request would exceed your organization’s rate limit of 80,000 input tokens per minute. For details, refer to: https://docs.anthropic.com/en/api/rate-limits; see the response headers for current usage. Please reduce the prompt length or the maximum tokens requested, or try again later. You may also contact sales at https://www.anthropic.com/contact-sales to discuss your options for a rate limit increase."}}
```

**Comment by any2names** [comment]
I have the same problem

**Comment by mkerciku** [comment]
I really hate this limitation and seems like its gonna persist for a while with Claude so if this gets implemented then you can leave claude do the tasks and just put a limit on total api requests and go do smth else. 1 api request per minute seems like its not causing anymore 429 error so I think this would be a good idea

**Comment by ForceAwakenz** [comment]
Really it is THE ONLY issue why I'm considering Cursor over VSCode + Cline

**Comment by patchshorts** [comment]
Plus one.

**Comment by SiNZeRo** [comment]
also the problem when using clline with siliconflow, 429 errors always show

**Comment by 417Goose** [comment]
can you increase the 80k rate limit?

**Comment by danielrosehill** [comment]
Anyone else finding that Anthropic is much tougher with rate limits than using Anthropic via OpenRouter? I hit the Anthropic rate limiting whenever I try to use it directly. With Open Router, I notice it gets _really_ slow some times which I assume is them implementing the rate-limiting behind the scenes.

**Comment by dcbartlett** [comment]
We just released a retry system ( thanks @ViezeVingertjes ) that should help alleviate this some.  It will automatically retry a request that errors due to rate limits (ones that return 429) after a period of time (increases with each retry).

Go get 3.3.0 for this feature :D  Happy Cline-ing

**Comment by 417Goose** [comment]
That is wonderful, sometimes I just sit her waiting for time to pass, now I
can just go on to other things.
